### PR TITLE
Feature/csh 5299 complete schema

### PR DESCRIPTION
**Pull request by vitaly-retel**

### CODE REVIEW

File: lib/components-schema-v1_0_x.ts [hunk 1]
```diff
@@ -95,29 +99,57 @@ export const componentsDefinitionSchema_v1_0_x = {
                                     options: {
                                         type: 'array',
                                         minItems: 1,
+                                        items: {
+                                            type: 'object',
+                                            properties: {
+                                                caption: {
```
Comment (wwsno) [inline lib/components-schema-v1_0_x.ts]: Can you add descriptions for the new properties added?

Comment (vitaly-retel) [reply lib/components-schema-v1_0_x.ts]: sure

File: lib/components-types-v1_0_x.ts [hunk 2]
```diff
@@ -83,36 +83,62 @@ export interface ComponentsDefinitionV10X {
...
+          type: "select";
+          options: {
+            /**
+             * Label of the item
+             */
+            caption: string;
+            /**
+             * Value of the item
+             */
+            value?: string;
```
Comment (wwjhu) [inline lib/components-types-v1_0_x.ts]: Maybe good to add to the comment why you can leave the value out.

File: lib/components-types-v1_0_x.ts [hunk 3]
```diff
@@ -83,36 +83,62 @@ export interface ComponentsDefinitionV10X {
...
+           */
+          pattern?: string;
+          /**
+           * Default value which is used instead of empty value
+           */
+          defaultValue?: string;
+          /**
+           * Unit type
+           */
+          unit?: string;
```
Comment (wwjhu) [inline lib/components-types-v1_0_x.ts]: Add to the comment examples of units

File: lib/components-types-v1_0_x.ts [hunk 4]
```diff
@@ -83,36 +83,62 @@ export interface ComponentsDefinitionV10X {
...
           [k: string]: any;
         }
       | {
           /**
            * Color picker field property
            */
-          type?: "colorPicker";
-          [k: string]: any;
+          type: "colorPicker";
+          options?: {
```
Comment (wwjhu) [inline lib/components-types-v1_0_x.ts]: Why is the opacity in an options object while for other fields properties are siblings?

File: lib/components-types-v1_0_x.ts (hunk 2 above)
Comment (wwsno) [reply lib/components-types-v1_0_x.ts]: Do keep in mind this file is generated from lib/components-schema-v1_0_x.ts, so the changes should go in there.

File: lib/components-types-v1_0_x.ts (hunk 4 above)
Comment (wwsno) [reply lib/components-types-v1_0_x.ts]: Guess we could drop the options thing?

File: lib/components-schema-v1_0_x.ts [hunk 5]
```diff
@@ -95,29 +99,63 @@ export const componentsDefinitionSchema_v1_0_x = {
...
                                     },
+                                    pattern: {
+                                        type: 'string',
+                                        description: 'Value validation regexp pattern',
+                                    },
+                                    defaultValue: {
+                                        type: 'string',
+                                        description: 'Default value which is used instead of empty value',
+                                    },
+                                    unit: {
```
Comment (wwsno) [inline lib/components-schema-v1_0_x.ts]: Just wondering, is it a limited number of string options we support here? If so we could further restrict it?

**Pull request merged**